backend/icpy/agent/tools/websearch_tools.py

```
@@ -0,0 +1,121 @@
+				return ToolResult(success=False, error="TAVILY_API_KEY is not set in the environment")
+
+			max_results = kwargs.get("maxResults")
+			depth = kwargs.get("searchDepth")
+			include_answer = bool(kwargs.get("includeAnswer", True))
+
+			payload = self._build_payload(str(query), max_results, depth, include_answer)
+
+			# Perform the request
+			try:
+				resp = requests.post(self._endpoint, json=payload, timeout=20)
+			except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)